aten/src/ATen/native/DistributionTemplates.h

```diff
@@ -249,39 +215,40 @@ Tensor& normal_out_impl(Tensor& output, double mean, const Tensor& std, c10::opt
 template<template<typename> class normal_kernel, typename RNG>
 Tensor& normal_out_impl(Tensor& output, const Tensor& mean, const Tensor& std, c10::optional<Generator> gen) {
   CHECK_NORMAL_TENSOR_STD(std);
-  bool is_deprecated_th_impl = resize_output_for_normal(output, mean, std);
+  auto shape = at::infer_size(mean.sizes(), std.sizes());
+  at::native::resize_output(output, shape);
   normal_impl_<normal_kernel, RNG>(output, 0, 1, gen);
   // CUDA NB: addcmul_out copies the tensor to be added into the output.
   // Please look at aten/src/THC/generic/THCTensorMathPointwise.cu
   // The previous function here was addcmul_out(output, mean, output, std, 1);
   // The third argument is not a constant reference and hence the samples in output are overwritten.
   // Consequently, the computation performed is mean + mean * std instead of mean + output * std
-  if (is_deprecated_th_impl) {
-    output.mul_(std.reshape(mean.sizes())).add_(mean);
-  }
-  else {
-    output.mul_(std).add_(mean);
-  }
+  output.mul_(std).add_(mean);
   return output;
 }
 
 template<template<typename> class normal_kernel, typename RNG>
 Tensor normal_impl(const Tensor& mean, double std, c10::optional<Generator> gen) {
+  CHECK_NORMAL_STD(std);
   Tensor ret = at::empty_like(mean, MemoryFormat::Contiguous);
   normal_out_impl<normal_kernel, RNG>(ret, mean, std, gen);
   return ret;
 }
 
 template<template<typename> class normal_kernel, typename RNG>
 Tensor normal_impl(double mean, const Tensor& std, c10::optional<Generator> gen) {
+  CHECK_NORMAL_TENSOR_STD(std);
   Tensor ret = at::empty_like(std, MemoryFormat::Contiguous);
   normal_out_impl<normal_kernel, RNG>(ret, mean, std, gen);
   return ret;
 }
 
 template<template<typename> class normal_kernel, typename RNG>
 Tensor normal_impl(const Tensor& mean, const Tensor& std, c10::optional<Generator> gen) {
-  Tensor ret = at::empty({0}, mean.options(), MemoryFormat::Contiguous);
+  CHECK_NORMAL_TENSOR_STD(std);
+  Tensor ret = at::empty_like(mean, MemoryFormat::Contiguous);
+  auto shape = at::infer_size(mean.sizes(), std.sizes());
+  at::native::resize_output(ret, shape);
```

Comment:
We can just construct `ret` with the correct shape, instead of constructing it here and resizing. You can do that with `at::empty(shape,  mean.options(), MemoryFormat::Contiguous);

Reply:
sure, i guess that might be faster but then the code will look more ad-hoc (i'm not sure if it affects the behavior either). so i'm in favor of keeping things as is unless it's terribly wrong

Reply:
discussed this with mario, it was suggested that i use your version due to potential performance implications

Reply:
as for other comments related to resize, i stand by what i wrote and welcome you to clarify if necessary

Reply:
yeah perf was what I had in mind :) sounds good

Reply:
pushed a change that does this. pls take a look once ci succeeds